src/UI/Actions.hs

```diff
@@ -236,32 +270,58 @@ toggleHeaders = Action
       Filtered -> set (asMailView . mvHeadersState) ShowAll s
       ShowAll -> set (asMailView . mvHeadersState) Filtered s
 
+applyTags :: [Text] -> Action ctx AppState
+applyTags ts = Action "apply given tags" (\s -> liftIO . selectedMailHelper s $ \m -> applyMailTags ts m s)
+
+reloadMails :: Action 'BrowseMail AppState
+reloadMails = Action "reload list of mails" applySearch
+
 -- Function definitions for actions
 --
 applySearch :: AppState -> T.EventM Name AppState
 applySearch s =
    runExceptT (getMessages searchterms (view (asConfig . confNotmuch) s))
-   >>= pure . ($ s) . either setError reloadListOfMails
+   >>= pure . ($ s) . either setError updateMailsList
      where searchterms = currentLine $ view (asMailIndex . miSearchEditor . E.editContentsL) s
 
+selectedMailHelper
+    :: Applicative f
+    => AppState -> (NotmuchMail -> f (AppState -> AppState)) -> f AppState
```

Comment:
Not for now, but I see this as being generalisable to `selectedItemsHelper`.  The next time we need a helper to do something with a "selected item(s)", remember this :)

Reply:
Good idea actually.

Reply:
Good idea actually.